python/src/main.cpp

```diff
@@ -217,7 +217,26 @@ PYBIND11_MODULE(kp, m) {
             py::arg("spirv"),
             py::arg("workgroup") = kp::Workgroup(),
             py::arg("spec_consts") = kp::Constants(),
-            py::arg("push_consts") = kp::Constants());
+            py::arg("push_consts") = kp::Constants())
+        .def("get_device_properties", &kp::Manager::getDeviceProperties, "Return a struct containing information about the device");
```

Comment:
Is there a way to just return the device properties as a Dict instead of a new class? Conscious if we start exposing Vk classes it could lead into more vk classes being requested